src/main/java/com/github/manolo8/darkbot/modules/DisconnectModule.java

```diff
@@ -83,6 +92,8 @@ public void tickStopped() {
             else {
                 pauseUntil = System.currentTimeMillis() + pauseTime;
                 main.setRunning(false);
+
+                if (closeBot) System.exit(0);
```

Comment:
I see a few main lacking things here:
 - Whenever we call system.exit, we *always* log explaining why the bot closed, makes it easier to know  (just posting bot exited and the reason would do)
  - It'd be nice if the status in the module also clarified that the bot won't just pause, but completely exit when the countdown gets to 0
  - It'd be nice if, instead of instantly closing when reaching 0, it opened a pop-up telling the user the bot will automatically close in 10s (counting down) and closing only after that. That way the user can cancel the shutdown instead of having no control over it shutting down just because they missconfigured something and now the bot is instantly shutting down without them knowing, possibly on loop every time they turn the bot on

Reply:
Regarding the status, I think that the one who should make the warning is the one who uses the option by means of the variable "reason".
The option to add the pop-up can be interesting especially if the user has activated the "Confirm exit" option in the bot's configuration.